static/js/message_list_view.js

```diff
@@ -1330,7 +1330,7 @@ export class MessageListView {
         // is rendered.
         const needed_rerender = this.maybe_rerender();
 
-        if (needed_rerender) {
+        if (needed_rerender && narrow_state.is_message_feed_visible()) {
             render_info = {need_user_to_scroll: true};
         }
```

Comment:
This feels logically wrong though -- we only need the user to scroll if this message list is the one visible. I guess the caller is arranging to only use the `render_info` variable in the case that it is, but it feels pretty messy as an abstraction for `append`. I think I'm liking more the other proposal, where we rename this return value parameter to something that communicates "The newly arrived message is out of view" which may or may not require the user to scroll depending on whether this message list is currently visible.